src/linker/Linker/Driver.cs

```diff
@@ -689,8 +701,15 @@ protected int SetupContext (ILogger? customLogger = null)
 			if (!new_mvid_used && !deterministic_used) {
 				context.DeterministicOutput = true;
 			}
-			if (dumpDependencies)
-				AddXmlDependencyRecorder (context, dependenciesFileName);
+			if (dumpDependencies) {
+				if (fileType == DependenciesFileFormat.Xml) {
+					AddXmlDependencyRecorder (context, dependenciesFileName);
+				} else {
```

Comment:
Nit: I would turn this into a switch - with Xml, Dgml cases and you can avoid the default with the StaticCS thing Andy added. It's more robust - if we add a new option things will not fall back to wrong branches that way.

Reply:
makes sense!